core/src/main/scala/com/salesforce/op/stages/impl/tuning/DataCutter.scala

```diff
@@ -82,33 +85,87 @@ class DataCutter(uid: String = UID[DataCutter]) extends Splitter(uid = uid) with
    * @param data
    * @return Parameters set in examining data
    */
-  override def preValidationPrepare(data: Dataset[Row]): Option[SplitterSummary] = {
-
-    import data.sparkSession.implicits._
-
-    val labels = data.map(r => r.getDouble(0) -> 1L)
-    val labelCounts = labels.groupBy(labels.columns(0)).sum(labels.columns(1)).persist()
-    val (resKeep, resDrop) = estimate(labelCounts)
-    labelCounts.unpersist()
-    setLabels(resKeep, resDrop)
+  override def preValidationPrepare(data: Dataset[Row], labelColNameOpt: Option[String]): Option[SplitterSummary] = {
+    val labelColName = labelColNameOpt.getOrElse(data.columns(0))
+    val labelColIdx = data.columns.indexOf(labelColName)
```

Comment:
I wouldnt assume that the first column is the label - that is the problem you had in the first place. Make a spark parameter on the base class where you set the name of the label column

Reply:
you == current version? it's now a method parameter (which I could make a DC param).

Reply:
I thought the rearranging of the dataframe was to solve this ordering assumption...